packages/client/hmi-client/src/components/widgets/tera-infinite-canvas.vue

```diff
@@ -142,6 +157,26 @@ onMounted(() => {
 		svg.transition().call(zoom.transform as any, d3.zoomIdentity);
 	}
 });
+
+function drawNewEdge(): string {
+	if (newEdge.value) {
+		return drawEdge(newEdge.value);
+	}
+	return 'M0,0';
+}
+
+function drawEdge(edge: WorkflowEdge): string {
+	if (edge.points && edge.points.length === 2) {
+		const sourcePoint = edge.points[0];
+		const targetPoint = edge.points[1];
+		const path = d3.path();
+		path.moveTo(sourcePoint.x, sourcePoint.y);
+		path.lineTo(targetPoint.x, targetPoint.y);
+		path.closePath();
+		return path.toString();
+	}
+	return 'M0,0';
+}
```

Comment:
There are better ways to do this since you are already hooked into D3's system. D3 has generator functions.

```
// The generator function, leave out the curve part if you don't want splines
const pathFn = d3
	.line<{ x: number; y: number }>()
	.x((d) => d.x)
	.y((d) => d.y)
	.curve(d3.curveBasis);
```

then

```
:d="pathFn(edge.points)"
```